File: solutions/python/algorithm/binarytree/Exercise2.py
from structure.dynamic.DynamicBinaryTreeADT import DynamicBinaryTreeADT


def count_nodes(tree):
    if tree is None or tree.isEmpty():
        return 0
    return 1 + count_nodes(tree.getLeft()) + count_nodes(tree.getRight())


# Ejemplo de ejecución
tree = DynamicBinaryTreeADT()
tree.add(10)
tree.add(5)
tree.add(15)

print("Cantidad de nodos:", count_nodes(tree))  # Salida: 3

File: solutions/javascript/src/algorithm/binarytree/Exercise2.js
import DynamicBinaryTreeADT from "../../structure/dynamic/DynamicBinaryTreeADT.js";

function countNodes(tree) {
  if (!tree || tree.isEmpty()) {
    return 0;
  }
  return 1 + countNodes(tree.getLeft()) + countNodes(tree.getRight());
}


// Ejemplo de ejecución
const tree = new DynamicBinaryTreeADT();
tree.add(10);
tree.add(5);
tree.add(15);

console.log("Cantidad de nodos:", countNodes(tree)); // Salida: 3